crates/sound/src/audio_backends/pipewire.rs

```diff
@@ -411,13 +430,20 @@ impl AudioBackend for PwBackend {
                         let streams = streams
                             .get_mut(stream_id as usize)
                             .expect("Stream does not exist");
+
                         let Some(buffer) = streams.buffers.front_mut() else {
                             return;
                         };
 
+                        // only for requests that have arrived after START
+                        if buffer.arrival_req_time.elapsed() < streams.period && buffer.stream_state == PCMState::Start {
+                            // TODO: replace this with silence
```

Comment:
I don't understand why you intend to use silence here after Start state?

Reply:
That comment is because I am not sure if it is OK to sleep in the threadloop. I was wondering if it is better to just fill a buffer with silent instead of sleeping.

Reply:
I am not sure about using silence as a preferred replacement either. because IIUC from the PR description, the idea is for the device to induce a delay right? which you already handled in pipewire.rs:L456 hence, I'll suggest to remove this conditional block of code. except I'm missing something?

Reply:
With that condition block, I induce a delay that is about a period. Otherwise, I do not how much delay I am adding. Another solution could be: 
1. to remove that conditional block
2. add a log if situation arrives
3. in the future, if we observe that the delay is not enough, we review the approach.

Reply:
By thinking again about this, we can remove that conditional block if we are sure that pw is calling process() in the right moment. Before, we were reading from guest's memory before process() was invoked. Now, we read from guest in the right moment which is when process() is invoked by pw.




[12:50](https://redhat-internal.slack.com/archives/C056HKK500Y/p1694688600793789)
before, we were reading from guest memory before process() was invoked. Now, we read from guest in the right moment which is when process() is invoked by pw

Reply:
IMHO, i don't think this delay is needed

Reply:
I think you are right. I will remove it for the moment. Thanks.